### DISCUSSION TITLE
How to check the original doc comment if I used `<inheritdoc/>`?

### DISCUSSION THREAD

**Discussion by kyoyama-kazusa** [dotnet/roslyn#53172] [Q&A] [answered]
Hi everyone.

I've been implementing a function that outputs the doc comments in a solution. However I can't figure out how to find the original doc comment when a certain member is marked `<inheritdoc/>`.

For example,

```csharp
/// <inheritdoc/>
public IEnumerator<int> GetEnumerator()
{
    // ...
}
```

The doc comment will be located and copied from the doc comment of the member `IEnumerable<T>.GetEnumerator`.

By the way... Some doc comments are in the built-in types, so I can't extract them. How can I get them?

Thank you. :smile:

**Comment by sharwell** [comment]
Can you give an example for the specific steps you wanted to follow to get this information? This will help me frame the answer in the correct context. 😄

**Reply by kyoyama-kazusa** [reply]
For example, I wrote a `struct` that contains a method `ToString(string?, IFormatProvider?)`:

```csharp
public struct Struct : IFormattable
{
    ...

    /// <inheritdoc/>
    public readonly string ToString(string? format, IFormatProvider? formatProvider) => ...

    ...
}
```

When I move the cursor onto that member `ToString`, tips shows the details like:

```
Formats the value of the current instance using the specified format.
Returns:
  The value of the current instance in the specified format.
```

![](https://user-images.githubusercontent.com/23616315/117223341-0500a380-ae40-11eb-9959-9d51acbf15fd.png)

However, I don't know how this tip got this detail information.

I know that this information binds with the method `IFormattable.ToString(string?, IFormatProvider?)`, but... I still don't know how I can get them.

😢

**Comment by MartyIX** [comment, marked as answer]
(May be relevant: https://github.com/dotnet/roslyn/discussions/50192)

**Reply by kyoyama-kazusa** [reply]
Thank you so much!
